src/app/core/export/query.service.spec.ts

```diff
@@ -635,6 +635,30 @@ describe("QueryService", () => {
     expect(res).toBe(3);
   });
 
+  it("should calculate the average of values", () => {
+    const data = [
+      { a: 5, b: 2 },
+      { b: "11" },
+      { b: -1 },
+      { b: 100 },
+      { a: "invalid" },
+    ];
+
+    let res = service.queryData("a:avg", undefined, undefined, data);
+    expect(res).toBe("5");
+
+    // Returns valid number if no values are available
+    res = service.queryData("a:avg", undefined, undefined, [{ b: 1 }]);
+    expect(res).toBe("0");
+
+    // Numbers are fixed to provided decimals
+    res = service.queryData("a:avg(2)", undefined, undefined, [
+      { a: 3 },
+      { a: 2 },
+    ]);
+    expect(res).toBe("2.50");
```

Comment:
let's add a test with one entry `= 0` (I agree that missing and invalid should be ignored, but 0 should be included in calculation)